praw/models/reddit/base.py

```diff
@@ -56,41 +79,48 @@ def __init__(self, reddit, _data):
         :param reddit: An instance of :class:`~.Reddit`.
 
         """
-        super(RedditBase, self).__init__(reddit, _data)
+        super(RedditBase, self).__init__(reddit, _data=_data)
         self._fetched = False
         self._info_params = {}
 
     def __repr__(self):
-        """Return an object initialization representation of the instance."""
-        return '{}({}={!r})'.format(self.__class__.__name__, self.STR_FIELD,
-                                    str(self))
+        """Return a representation of the instance."""
+        return '<{}({}={!r})>'.format(self.__class__.__name__,
+                                      self.REPR_FIELD,
+                                      getattr(self, self.REPR_FIELD))
 
     def __str__(self):
         """Return a string representation of the instance."""
         return getattr(self, self.STR_FIELD)
 
-    def __ne__(self, other):
-        """Return whether the other instance differs from the current."""
-        return not self == other
-
     def _fetch(self):
         if '_info_path' in dir(self):
             other = self._reddit.get(self._info_path(),
                                      params=self._info_params)
         else:
             self._info_params['id'] = self.fullname
-            children = self._reddit.get(API_PATH['info'],
-                                        params=self._info_params).children
+            children = self._reddit \
+                .get(API_PATH['info'],
+                     params=self._info_params)._data['children']
             if not children:
                 raise PRAWException('No {!r} data returned for thing {}'
                                     .format(self.__class__.__name__,
                                             self.fullname))
             other = children[0]
-        self.__dict__.update(other.__dict__)
+        self._data.update(other._data)
         self._fetched = True
 
     def _reset_attributes(self, *attributes):
         for attribute in attributes:
-            if attribute in self.__dict__:
-                del self.__dict__[attribute]
+            if attribute in self._data:
+                del self._data[attribute]
         self._fetched = False
+
+    def fetch(self, force=False):
```

Comment:
> What do you propose I do with it?

I propose we remove it.

Reply:
This attribute intends to give users more control over PRAW’s models. There are some use cases for `fetch()` that we currently don’t have any recommended equivalents for at the moment.

For instance, there may be cases where users may not need to target a specific attribute but would still like to trigger a fetch. There are two ways of triggering a fetch in this scenario:

1. Use `praw_object._fetch()`, which is supposedly meant to be private.
2. Do something strange like the following:

```python3
try:
	praw_object.name
except AttributeError:
	pass
```

The second approach is not clear about what it’s doing and requires much more typing than `_fetch()`. If available, it would be better to use `fetch()` instead since underscore members are supposed to be for internal use.

If the reason you don’t want to introduce `fetch()` and `fetched` at this time is because these changes will be inconsistent with previous versions, then please reconsider on the grounds that this PR is not meant to be backwards compatible.